Components/Sensors/BNO055.cs

```diff
@@ -0,0 +1,333 @@
+            BNO055_CALIB_STAT_ADDR = 0X35,
+            BNO055_SELFTEST_RESULT_ADDR = 0X36,
+            BNO055_INTR_STAT_ADDR = 0X37,
+
+            BNO055_SYS_CLK_STAT_ADDR = 0X38,
+            BNO055_SYS_STAT_ADDR = 0X39,
+            BNO055_SYS_ERR_ADDR = 0X3A,
+
+            /* Unit selection register */
+            BNO055_UNIT_SEL_ADDR = 0X3B,
+            BNO055_DATA_SELECT_ADDR = 0X3C,
+
+            /* Mode registers */
+            BNO055_OPR_MODE_ADDR = 0X3D,
+            BNO055_PWR_MODE_ADDR = 0X3E,
+
+            BNO055_SYS_TRIGGER_ADDR = 0X3F,
+            BNO055_TEMP_SOURCE_ADDR = 0X40,
+
+            /* Axis remap registers */
+            BNO055_AXIS_MAP_CONFIG_ADDR = 0X41,
+            BNO055_AXIS_MAP_SIGN_ADDR = 0X42,
+
+            /* SIC registers */
+            BNO055_SIC_MATRIX_0_LSB_ADDR = 0X43,
+            BNO055_SIC_MATRIX_0_MSB_ADDR = 0X44,
+            BNO055_SIC_MATRIX_1_LSB_ADDR = 0X45,
+            BNO055_SIC_MATRIX_1_MSB_ADDR = 0X46,
+            BNO055_SIC_MATRIX_2_LSB_ADDR = 0X47,
+            BNO055_SIC_MATRIX_2_MSB_ADDR = 0X48,
+            BNO055_SIC_MATRIX_3_LSB_ADDR = 0X49,
+            BNO055_SIC_MATRIX_3_MSB_ADDR = 0X4A,
+            BNO055_SIC_MATRIX_4_LSB_ADDR = 0X4B,
+            BNO055_SIC_MATRIX_4_MSB_ADDR = 0X4C,
+            BNO055_SIC_MATRIX_5_LSB_ADDR = 0X4D,
+            BNO055_SIC_MATRIX_5_MSB_ADDR = 0X4E,
+            BNO055_SIC_MATRIX_6_LSB_ADDR = 0X4F,
+            BNO055_SIC_MATRIX_6_MSB_ADDR = 0X50,
+            BNO055_SIC_MATRIX_7_LSB_ADDR = 0X51,
+            BNO055_SIC_MATRIX_7_MSB_ADDR = 0X52,
+            BNO055_SIC_MATRIX_8_LSB_ADDR = 0X53,
+            BNO055_SIC_MATRIX_8_MSB_ADDR = 0X54,
+
+            /* Accelerometer Offset registers */
+            ACCEL_OFFSET_X_LSB_ADDR = 0X55,
+            ACCEL_OFFSET_X_MSB_ADDR = 0X56,
+            ACCEL_OFFSET_Y_LSB_ADDR = 0X57,
+            ACCEL_OFFSET_Y_MSB_ADDR = 0X58,
+            ACCEL_OFFSET_Z_LSB_ADDR = 0X59,
+            ACCEL_OFFSET_Z_MSB_ADDR = 0X5A,
+
+            /* Magnetometer Offset registers */
+            MAG_OFFSET_X_LSB_ADDR = 0X5B,
+            MAG_OFFSET_X_MSB_ADDR = 0X5C,
+            MAG_OFFSET_Y_LSB_ADDR = 0X5D,
+            MAG_OFFSET_Y_MSB_ADDR = 0X5E,
+            MAG_OFFSET_Z_LSB_ADDR = 0X5F,
+            MAG_OFFSET_Z_MSB_ADDR = 0X60,
+
+            /* Gyroscope Offset register s*/
+            GYRO_OFFSET_X_LSB_ADDR = 0X61,
+            GYRO_OFFSET_X_MSB_ADDR = 0X62,
+            GYRO_OFFSET_Y_LSB_ADDR = 0X63,
+            GYRO_OFFSET_Y_MSB_ADDR = 0X64,
+            GYRO_OFFSET_Z_LSB_ADDR = 0X65,
+            GYRO_OFFSET_Z_MSB_ADDR = 0X66,
+
+            /* Radius registers */
+            ACCEL_RADIUS_LSB_ADDR = 0X67,
+            ACCEL_RADIUS_MSB_ADDR = 0X68,
+            MAG_RADIUS_LSB_ADDR = 0X69,
+            MAG_RADIUS_MSB_ADDR = 0X6A
+        }
+
+        /// <summary>
+        /// The power mode of the BNO055. 
+        /// </summary>
+        private enum Powermode

```

Comment:
This should be `PowerMode`, to match with the style of `OperationMode` just below.

Also here, it's pretty redundant to have the enum name in the enum types. So instead of `POWER_MODE_LOWPOWER`, maybe make it `LowPower`.

Reply:
Sorry, I should be consistent with the style here. Not `LowPower`, but `LOW_POWER`.